packages/service-library/src/servicelib/long_running_tasks/_redis_store.py

```diff
@@ -134,7 +136,14 @@ async def mark_for_removal(self, task_id: TaskId) -> None:
         await handle_redis_returns_union_types(
             self._redis.hset(
                 self._get_redis_task_data_key(task_id),
-                mapping=_to_redis_hash_mapping({_MARKED_FOR_REMOVAL_FIELD: True}),
+                mapping=_to_redis_hash_mapping(
+                    {
+                        _MARKED_FOR_REMOVAL_FIELD: True,
+                        _MARKED_FOR_REMOVAL_AT_FIELD: datetime.datetime.now(
```

Comment:
Q: why not have an optional `_MARKED_FOR_REMOVAL_AT_FIELD` field? if it is null it means there are no such thing. If it exists it means you want to remove it.. no need for the additional boolean.

Reply:
Good point, managed to removed the extra field here